File: debate-agent/frontend/src/api/think.ts
/**
 * Think engine API functions.
 */

import { api } from "./client";

export interface ThinkCreate {
  topic: string;
  depth?: "quick" | "standard" | "deep";
  models?: string[];
}

export interface ThinkMessageResponse {
  id: string;
  task_id: string;
  model_id?: string;
  role: string;
  round_number: number;
  content: string;
  structured?: Record<string, unknown>;
  created_at?: string;
}

export interface ThinkResponse {
  id: string;
  topic: string;
  status: string;
  depth?: string;
  created_at?: string;
  completed_at?: string;
  synthesis?: string;
  insights?: string[];
  messages?: ThinkMessageResponse[];
}

export interface ThinkListResponse {
  sessions: ThinkResponse[];
  total: number;
  page: number;
  page_size: number;
}

export const thinkApi = {
  create: (data: ThinkCreate) => api.post<ThinkResponse>("/think", data),
  list: (page = 1, pageSize = 10) =>
    api.get<ThinkListResponse>(`/think?page=${page}&page_size=${pageSize}`),
  get: (id: string) => api.get<ThinkResponse>(`/think/${id}`),
  start: (id: string) => api.post<ThinkResponse>(`/think/${id}/start`),
};

File: debate-agent/backend/schemas/think.py
"""
Think engine Pydantic schemas.
"""

from typing import List, Optional, Dict
from datetime import datetime
from pydantic import BaseModel, Field


class ThinkCreate(BaseModel):
    """Schema for creating a think session."""
    topic: str = Field(..., min_length=1, max_length=500, description="思考主题")
    depth: str = Field(default="standard", description="思考深度: quick/standard/deep")
    models: List[str] = Field(default_factory=lambda: ["mimo"], description="参与模型列表")
    config: Optional[Dict] = Field(default=None, description="额外配置")


class ThinkResponse(BaseModel):
    """Schema for think session response."""
    id: str
    topic: str
    status: str
    depth: Optional[str] = None
    created_at: Optional[str] = None
    completed_at: Optional[str] = None
    synthesis: Optional[str] = None
    insights: Optional[List[str]] = None
    messages: Optional[List["ThinkMessageResponse"]] = None

    class Config:
        from_attributes = True


class ThinkMessageResponse(BaseModel):
    """Schema for think message response."""
    id: str
    task_id: str
    model_id: Optional[str] = None
    role: str
    round_number: int
    content: str
    structured: Optional[Dict] = None
    created_at: Optional[str] = None

    class Config:
        from_attributes = True


class ThinkListResponse(BaseModel):
    """Schema for think list response."""
    sessions: List[ThinkResponse]
    total: int
    page: int
    page_size: int
